tasty-query/shared/src/main/scala/tastyquery/Subtyping.scala

```diff
@@ -0,0 +1,306 @@
+
+    case tp2: WildcardTypeBounds =>
+      isSubtype(tp1, tp2.bounds.high)
+
+    case tp2: ThisType =>
+      val cls2 = tp2.cls
+      tp1 match
+        case tp1: ThisType =>
+          tp1.cls == cls2
+        case tp1: NamedType if cls2.is(Module) && cls2 == tp1.symbol =>
+          (cls2.isStatic || isSubprefix(tp1.prefix, cls2.typeRef.prefix))
+            || level2(tp1, tp2)
+        case _ =>
+          level2(tp1, tp2)
+
+    case tp2: AndType =>
+      isSubtype(tp1, tp2.first) && isSubtype(tp1, tp2.second)
+
+    case tp2: ConstantType =>
+      tp1 match
+        case tp1: ConstantType =>
+          tp1.value == tp2.value
+        case _ =>
+          level2(tp1, tp2)
+
+    case _ =>
+      level2(tp1, tp2)
+  end level1
+
+  private def level1NamedNamed(tp1: NamedType, tp2: NamedType)(using Context): Boolean =
+    val sym1 = tp1.symbol
+    val sym2 = tp2.symbol
+
+    if sym1 == sym2 then
+      sym1.isStatic
+      || isSubprefix(tp1.prefix, tp2.prefix)
+      || level2(tp1, tp2)
+    else
+      /* When the symbols are not the same, we could still have a subtyping
+       * relationship if they are defined in different classes of the same hierarchy.
+       */
+      val trueBecauseOverriddenMembers =
+        sym1.name == sym2.name
+          && isSubprefix(tp1.prefix, tp2.prefix)
+          && tp1.optSignature == tp2.optSignature
+          && !(sym1.isClass && sym2.isClass) // classes can shadow each other without being subtypes
+      trueBecauseOverriddenMembers || level2(tp1, tp2)
+  end level1NamedNamed
+
+  private def level2(tp1: Type, tp2: Type)(using Context): Boolean = tp1 match
+    case tp1: NamedType =>
+      val sym1 = tp1.symbol
+      sym1 match
+        case sym1: TypeMemberSymbol if sym1.isTypeAlias =>
+          if isSubtype(sym1.aliasedTypeAsSeenFrom(tp1.prefix), tp2) then true
+          else level3(tp1, tp2)
+        case _ =>
+          (sym1 == defn.NothingClass || isBottom(tp1))
+            || level3(tp1, tp2)
+      end match
+
+    case tp1: ThisType =>
+      val cls1 = tp1.cls
+      tp2 match {
+        case tp2: TermRef if cls1.is(Module) && cls1.eq(tp2.typeSymbol) =>
+          (cls1.isStatic || isSubprefix(cls1.typeRef.prefix, tp2.prefix))
+            || level3(tp1, tp2)
+        case _ =>
+          level3(tp1, tp2)
+      }
+
+    case tp1: WildcardTypeBounds =>
+      isSubtype(tp1.bounds.low, tp2)
+
+    case tp1: OrType =>
+      isSubtype(tp1.first, tp2) && isSubtype(tp1.second, tp2)
+
+    case tp1: AnnotatedType =>
+      isSubtype(tp1.typ, tp2)
+
+    case _ =>
+      level3(tp1, tp2)
+  end level2
+
+  private def level3(tp1: Type, tp2: Type)(using Context): Boolean = tp2 match
+    case tp2: TypeRef =>
+      tp2.symbol match
+        case cls2: ClassSymbol =>
+          if cls2.typeParams.isEmpty then
+            if isBottom(tp1) then true
+            else if tp1.isLambdaSub then false // should be tp1.hasHigherKind, but the scalalib does not like that
```

Comment:
Does this comment apply to tasty-query too? :) (In fact, I wonder if the original comment is still true, there were some fixes in the standard library)